crates/consensus/src/transaction/eip4844/builder.rs

```diff
@@ -0,0 +1,390 @@
+#[cfg(not(feature = "kzg"))]
+use alloy_eips::eip4844::Blob;
+#[cfg(feature = "kzg")]
+use c_kzg::{Blob, KzgCommitment, KzgProof, KzgSettings};
+
+use alloy_eips::eip4844::{BYTES_PER_BLOB, FIELD_ELEMENTS_PER_BLOB};
+
+use super::utils::WholeFe;
+
+/// A builder for creating a [`BlobTransactionSidecar`].
+///
+/// [`BlobTransactionSidecar`]: crate::BlobTransactionSidecar
+#[derive(Debug, Clone)]
+pub struct PartialSidecar {
+    /// The blobs in the sidecar.
+    blobs: Vec<Blob>,
+    /// The number of field elements that we have ingested, total.
+    fe: usize,
+}
+
+impl Default for PartialSidecar {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PartialSidecar {
+    /// Create a new builder.
+    pub fn new() -> Self {
+        // NB: vecs default to 100 capacity. Blobs are large. We don't want
+        // to allocate 100 blobs if we don't need them.
```

Comment:
This is not the case, `Vec::new()` does not allocate capacity

Reply:
and on the first push it will allocate for 1: https://github.com/rust-lang/rust/blob/3cbb93223f33024db464a4df27a13c7cce870173/library/alloc/src/raw_vec.rs#L130-L141